src/android/toga_android/app.py

```diff
@@ -7,9 +7,21 @@ class MainWindow(Window):
     pass
 
 
-class TogaApp:
+# The `IPythonApp` interface in Java allows Python code to
+# run on Android activity lifecycle hooks such as `onCreate()`.
+IPythonApp = JavaInterface('org/beeware/android/IPythonApp')
```

Comment:
This should probably move to a new `libs` package that contains all the declared bindings to the Android system libraries (see the cocoa/iOS backends for the analog here)